beacon-chain/blockchain/head.go

```diff
@@ -134,31 +134,27 @@ func (s *Service) saveHead(ctx context.Context, headRoot [32]byte) error {
 // root in DB. With the inception of inital-sync-cache-state flag, it uses finalized
 // check point as anchors to resume sync therefore head is no longer needed to be saved on per slot basis.
 func (s *Service) saveHeadNoDB(ctx context.Context, b *ethpb.SignedBeaconBlock, r [32]byte) error {
+	if featureconfig.Get().NewStateMgmt {
```

Comment:
For new state mgmt, there's no need to save head during initial syncing. There's no fork choice during initial syncing since a node is just syncing using a block list